ui/src/app/domains/views/DomainDetails/DomainDetailsHeader/AddRecordDomainForm/AddRecordDomainForm.tsx

```diff
@@ -0,0 +1,132 @@
+import { useCallback } from "react";
+
+import { Col, Row, Select } from "@canonical/react-components";
+import { useDispatch, useSelector } from "react-redux";
+import { useParams } from "react-router";
+import * as Yup from "yup";
+import type { SchemaOf } from "yup";
+
+import FormikField from "app/base/components/FormikField";
+import FormikForm from "app/base/components/FormikForm";
+import type { RouteParams } from "app/base/types";
+import { actions as domainActions } from "app/store/domain";
+import domainSelectors from "app/store/domain/selectors";
+import type { DomainResource } from "app/store/domain/types/base";
+import { RecordType } from "app/store/domain/types/base";
+
+const SelectValues = Object.values(RecordType).map((value) => {
```

Comment:
By convention in JavaScript, capitalized variables are reserved for instantiable classes, React components, and Types :)